### PR TITLE
[Ready]Adds the Stray Cargo pod event

### PR DESCRIPTION
**Pull request by XDTM**
## About The Pull Request

Adds an event where a cargo pod containing a random cargo supply pack crash-lands on a random spot on the station.
Added a rare variant (post 30 minutes) that spawns a null crate with 30 TCs of gear.

Crates which would be locked when ordered from cargo will start unlocked when falling with a stray cargo pod.

<s>_Note: i'm not yet able to compile or test the code myself, will do it as soon as possible, before tomorrow_</s> done, all fine

## Why It's Good For The Game

A fairly neutral event that could benefit some lucky assistant, throw shotguns inside the cult base, or simply land on the head of some very unlucky person.

I think more variety will be nice for the game.

## Changelog
:cl: XDTM
add: Added the Stray Cargo Pod event, which drops a pod containing a random cargo crate into a random location. Rarely, one might even contain assorted syndicate gear!
/:cl:

### CODE REVIEW

File: code/modules/events/stray_cargo.dm [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+		var/list/safe_area_types = typecacheof(list(
+		/area/ai_monitored/turret_protected/ai,
+		/area/ai_monitored/turret_protected/ai_upload,
+		/area/engine,
+		/area/shuttle)
+		)
+
+		///Subtypes from the above that actually should explode.
+		var/list/unsafe_area_subtypes = typecacheof(list(/area/engine/break_room))
+		allowed_areas = make_associative(GLOB.the_station_areas) - safe_area_types + unsafe_area_subtypes
```
Comment (Qustinnus) [inline code/modules/events/stray_cargo.dm]: never knew this proc existed but this is now also the only thing that uses it :^)

Comment (XDTM) [reply code/modules/events/stray_cargo.dm]: i copied the whole proc from the anomaly event so it's at the very least two things

File: code/modules/events/stray_cargo.dm [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+		CRASH("No valid areas for cargo pod found.")
+	var/list/turf_test = get_area_turfs(impact_area)
+	if(!turf_test.len)
+		CRASH("Stray Cargo Pod : No valid turfs found for [impact_area] - [impact_area.type]")
+
+///Spawns a random supply pack, puts it in a pod, and spawns it on a random tile of the selected area
+/datum/round_event/stray_cargo/start()
+	var/turf/LZ = pick(get_area_turfs(impact_area))
+	var/datum/supply_pack/SP = pick(SSshuttle.supply_packs)
+	var/obj/structure/closet/crate/C = SP.generate(null)
```
Comment (MrDoomBringer) [inline code/modules/events/stray_cargo.dm]: Just a heads up, you dont need to generate the supplypack here. Just pass the SP as the third parameter to the DPtarget call - the DPTarget's init functions handles SP generation.

**Comment by TiviPlus** [comment]
Time to watch the clown get hit by a SM shard crate in the halls


Seriously though SM crates could either be a hilarious event or griff central

**Comment by PKPenguin321** [comment]
Probably of what crate you get could scale off of the price of the crate
1/10000 chance for a bicycle to just sorta fall on you

**Comment by GravitronBlue** [comment]
Despite the flak I'll get for this suggestion, nullcrates should be reimplented for exclusive use by this event along with using the syndicate drop pod variant for them as well.

File: code/modules/events/stray_cargo.dm (hunk 2 above)
Comment (XDTM) [reply code/modules/events/stray_cargo.dm]: i saw that it can handle supply orders but not supply packs, unless i missed something

**Comment by XDTM** [comment]
there could definitely be a rarer variant with null crates

**Comment by XDTM** [comment]
added an event variant with null crates

**Comment by Spepbuthewranglesgithub** [comment]
bit of a boring event considering all the _fun_ stuff is going to be in locked crates, but ah well.

**Comment by MrDoomBringer** [comment]
Yeah, it should make the crates be unlocked - its a rare event after all, i think you can be a little generous with it

File: code/modules/events/stray_cargo.dm (hunk 2 above)
Comment (MrDoomBringer) [reply code/modules/events/stray_cargo.dm]: haha lol i'm retarded
its not like i wrote the drop pod code or anything
you're right, your implementation is good

**Comment by XDTM** [comment]
Secure crates now start unlocked when dropped

**Comment by cacogen** [comment]
tynan wants a word

**Comment by Qustinnus** [comment]
toady one wants one with Tynan

File: code/modules/events/stray_cargo.dm [hunk 3]
```diff
@@ -85,3 +86,4 @@
 
 /datum/round_event/stray_cargo/syndicate
 	possible_pack_types = list(/datum/supply_pack/misc/syndicate)
+	pod_type = /obj/structure/closet/supplypod/syndicate
```
Comment (MrDoomBringer) [inline code/modules/events/stray_cargo.dm]: You shouldn't create a new supplypod subtype - we already have a system to handle this. Just call setStyle(STYLE_SYNDICATE) on an initialized supplypod object. That function will handle icons, descriptions, etc.
```suggestion
	pod = new /obj/structure/closet/supplypod()
	pod.setStyle(STYLE_SYNDICATE)
```

**Review by MrDoomBringer — CHANGES_REQUESTED** [review]

File: code/modules/events/stray_cargo.dm (hunk 3 above)
Comment (MrDoomBringer) [reply code/modules/events/stray_cargo.dm]: You can find the styles you can use, etc. in the cargo.dm defines file

Comment (XDTM) [reply code/modules/events/stray_cargo.dm]: thanks

**Comment by optimumtact** [comment]
move the rare event to post 45 please.

**Pull request merged**